autogen/agentchat/conversable_agent.py

```diff
@@ -1058,6 +1095,53 @@ async def a_generate_oai_reply(
             None, functools.partial(self.generate_oai_reply, messages=messages, sender=sender, config=config)
         )
 
+    def _generate_code_execution_reply_using_executor(
```

Comment:
How is compatibility between the code producer agent and the code executor agent ensured in runtime (especially in groupchat which maybe contains multiple code executor)?

Reply:
This is a great question. I believe in this case the code executor agent must update its description to include what type of code execution interface it expects. 

We can update the `Agent.description` field for this.

Reply:
yes

Reply:
Actually, I am a bit hesitant to over-engineer this right now regarding to compatibility between agents.

Reply:
The current way of using capability and `add_to_agent` is a short-cut for users to use some default prompts. But I can imagine many users may want to customize their agent's system prompt and description anyway. I am thinking we want to avoid over-designing how system prompt and agent's description field are customized at the moment. So perhaps we can leave this for a future PR.

Reply:
No problem!